pkg/api/store/cluster/cluster_store.go

```diff
@@ -434,3 +435,55 @@ func setPrivateRegistryPasswordIfNotExists(oldData, newData map[string]interface
 	}
 	values.PutValue(newData, updatedConfig, "rancherKubernetesEngineConfig", "privateRegistries")
 }
+
+func setCloudProviderPasswordFieldsIfNotExists(oldData, newData map[string]interface{}) {
+	replaceWithOldSecretIfNotExists(oldData, newData, "openstackCloudProvider", "rancherKubernetesEngineConfig", "cloudProvider", "openstackCloudProvider", "global", "password")
```

Comment:
nit: do you think putting these fields in a map[cloudProviderName][]string outside the function would make it cleaner?

Reply:
Yeah earlier I was using that approach, but then the value for each cloudProvider key would be of varying length, meaning for Azure we have two secret fields, but for vsphere and openstack we first have a field called Global, and then its subfield is the password field. VCenter has a field which is a map of string to VirtualCenterConfig, which finally has the password field. So I'd have to have nested for loops still not making it any cleaner